docs/fundamentals/configuration.rst

```diff
@@ -1,8 +1,199 @@
+To work with settings in your ASP.NET application, you can instantiate a new instance of ``Configuration`` anywhere you need one. However, it's recommended to do this once in your application's ``Startup``, and then inject an instance of `IConfiguration <https://github.com/aspnet/Configuration/blob/dev/src/Microsoft.Framework.Configuration.Abstractions/IConfiguration.cs>`_ into any controllers or services that need to access configuration. At its simplest, the ``Configuration`` class is just a collection of ``Sources``, which provide the ability to read and write name/value pairs. You must configure at least one source in order for ``Configuration`` to function correctly. For instance, you could include the following code in any method in your ASP.NET application:
+
+.. code-block:: c#
+	:linenos:
+
+	// assumes using Microsoft.Framework.ConfigurationModel is specified
+	var config = new Configuration();
+	config.Add(new MemoryConfigurationSource());
+		
+	config.Set("somekey", "somevalue");
+	
+	// do some other work
+	
+	string setting = config.Get("somekey); // returns "somevalue"
+	// or
+	string setting2 = config["somekey"]; // also returns "somevalue"
+
+.. note:: You must set at least one configuration source.
+
+It's not unusual to store configuration values in a hierarchical structure, especially when using external files (e.g. JSON, XML, INI). In this case, configuration values can be retrieved using a ``:`` separated key, starting from the root of the hierarchy. For example, consider the following ``config.json`` file:
+
+.. _config-json:
+
+.. literalinclude:: configuration/sample/src/ConfigDemo/config.json
+	:linenos:
+	:language: javascript
+
+Access to the "SiteTitle" setting is done through a key of ``AppSettings:SiteTitle``. Similarly, access to the ``ConnectionString`` setting is achieved through this key: ``Data:DefaultConnection:ConnectionString``.
+
+Instantiating ``Configuration`` as shown above should be avoided. The recommended approach is to create your own settings class (probably several different classes, corresponding to different cohesive groups of settings), and use  :ref:`Options and configuration objects <options-config-objects>` and :doc:`dependency-injection` to access these classes wherever they're needed.
+
+Injecting the Configuration object
```

Comment:
Actually, I talked this over with @glennc and I don't think we want to be telling people to put config in DI at all - we recommend against that and folks should just use the options pattern instead. Let's remove this section.